plugins/obs-ffmpeg/obs-nvenc.c

```diff
@@ -1155,71 +1437,104 @@ static void *nvenc_create_internal(enum codec_type codec, obs_data_t *settings,
 }
 
 static void *nvenc_create_base(enum codec_type codec, obs_data_t *settings,
-			       obs_encoder_t *encoder)
+			       obs_encoder_t *encoder, bool texture)
 {
 	/* this encoder requires shared textures, this cannot be used on a
 	 * gpu other than the one OBS is currently running on. */
 	const int gpu = (int)obs_data_get_int(settings, "gpu");
-	if (gpu != 0) {
+	if (gpu != 0 && texture) {
 		blog(LOG_INFO,
-		     "[obs-nvenc] different GPU selected by user, falling back to ffmpeg");
+		     "[obs-nvenc] different GPU selected by user, falling back "
+		     "to non-texture encoder");
 		goto reroute;
 	}
 
 	if (obs_encoder_scaling_enabled(encoder)) {
-		if (!obs_encoder_gpu_scaling_enabled(encoder)) {
+		if (!obs_encoder_gpu_scaling_enabled(encoder) && texture) {
 			blog(LOG_INFO,
-			     "[obs-nvenc] CPU scaling enabled, falling back to ffmpeg");
+			     "[obs-nvenc] CPU scaling enabled, falling back to"
+			     " non-texture encoder");
 			goto reroute;
 		}
 		blog(LOG_INFO, "[obs-nvenc] GPU scaling enabled");
```

Comment:
By adding the `&& texture` above, this GPU scaling message will print if CPU scaling is enabled and `texture == false`.